plugins/module_utils/_api/utils/utils.py

```diff
@@ -240,9 +240,23 @@ def convert_service_networks(
 
 
 def parse_repository_tag(repo_name: str) -> tuple[str, str | None]:
```

Comment:
This function is called in a lot of places, changing its semantics can introduce a lot of subtle bugs. I don't think it's a good idea to change the behavior of this function.

I think it's better to have proper parsing for images. I've started working on that a longer time ago, but then got interrupted and haven't continued it for a longer time. The original approach in Docker SDK for Python - where this code comes from - seems to be pretty broken, and causes a lot of problems in quite a few places in this collection (and likely in other users of that code).